### PR TITLE
Turbolinks and recaptcha

### PR DESCRIPTION
**Pull request by paulbaum**
https://github.com/ambethia/recaptcha/issues/342

as I used a monkey patch for that which works for us I'm not sure if it would work for others / doesn't destroy anything.

I'm happy for feedback and I don't think this solution is the best but it works

### CODE REVIEW

File: lib/recaptcha/helpers.rb [hunk 1]
```diff
@@ -205,6 +211,27 @@ def self.to_message(_key, default)
...
+      nonce_attr = " nonce='#{nonce}'" if nonce
+
+      <<-HTML
+        <script#{nonce_attr}>
+          // Define function so that we can call it again later if we need to reset it
+          // This executes reCAPTCHA and then calls our callback.
+          function #{recaptcha_v3_execute_function_name(action)}() {
+            grecaptcha.ready(function() {
+              grecaptcha.execute('#{site_key}', {action: '#{action}'}).then(function(token) {
+                //console.log('#{id}', token)
```
Comment (grosser) [inline lib/recaptcha/helpers.rb]: remove ?

**Comment by grosser** [comment]
- can you high-level explain how this works ? (maybe add to readme)
 - add a test ?

**Comment by paulbaum** [comment]
>     * can you high-level explain how this works ? (maybe add to readme)
> 
>     * add a test ?

It basically puts the name of the function in the url used for loading googles recaptcha library as an onload callback. This way we don't need to execute the function ourselves. We still need to define it though!

turbolinks attributes true switches to the function that works that way.
If you use turbolinks - the onload property should not work any more.

File: lib/recaptcha/helpers.rb (hunk 1 above)
Comment (paulbaum) [reply lib/recaptcha/helpers.rb]: as  this gets executed after the callback we must not remove it

Comment (grosser) [reply lib/recaptcha/helpers.rb]: I meant the console.log :)

Comment (paulbaum) [reply lib/recaptcha/helpers.rb]: sure. This is still in the original code too. I'll remove it in both cases

**Comment by grosser** [comment]
can you please add the turbolinks option to the readme too ?

**Comment by paulbaum** [comment]
> can you please add the turbolinks option to the readme too ?

done :)

**Comment by grosser** [comment]
awesome!

**Comment by grosser** [comment]
5.3.0

**Pull request merged**